toolkit/pythonit_toolkit/api/permissions.py

```diff
@@ -1,9 +1,52 @@
+from typing import Any
+
+import jwt
+from pythonit_toolkit.headers import SERVICE_JWT_HEADER
 from pythonit_toolkit.pastaporto.entities import Credential
+from pythonit_toolkit.pastaporto.tokens import decode_service_to_service_token
 from strawberry.permission import BasePermission
+from strawberry.types import Info
 
 
 class IsAuthenticated(BasePermission):
     message = "Not authenticated"
 
     def has_permission(self, source, info, **kwargs):
         return Credential.AUTHENTICATED in info.context.request.auth.scopes
+
+
+def IsService(allowed_callers: list[str], secret: str, service: str):
+    if not allowed_callers:
+        raise ValueError("No callers allowed specified")
+
+    if not secret:
+        raise ValueError("JWT secret cannot be empty")
+
+    if not service:
+        raise ValueError("Current service name cannot be empty")
+
+    secret = str(secret)
+
+    class _IsService(BasePermission):
```

Comment:
why redefining the class here?

Reply:
every time the function is called a new class is created, this is because in strawberry you can't have a permission with parameters in the constructor

Reply:
The `IsService` that you import is actually a function, this allows us to send parameters

Reply:
it will then be used like this: https://github.com/pythonitalia/pycon/pull/2470/files#diff-f484eb8ffccc13ba354f92786a410bb34829668243880b0d0e9d1b6909f2f163R16

Reply:
here: https://github.com/pythonitalia/pycon/pull/2473